specification/dartLangSpec.tex

```diff
@@ -3154,11 +3158,44 @@ \subsubsection{Generative Constructors}
 <initializerExpression> ::= <conditionalExpression> | <cascade>
 \end{grammar}
 
+\LMHash{}%
+It is a compile-time error if a \synt{functionExpression} occurs as a subterm
+(\ref{notation})
+of an \synt{initializerExpression} $e$,
+unless it occurs inside a subterm of $e$ which is of the form
+\syntax{`(' <expression> `)'}.
```

Comment:
HOWEVER (ahem), that's probably *also* not precise enough, now that I got to think about it.

We do allow some function expressions that are not inside parentheses, as long as they are inside *something*.
Being "inside" can be inside a collection literal or string interpolation, or occurring as a function call argument, which are still clearly delimited, but not by a parenthesized expression.

So, maybe we need a definition like:

> A subexpression *e* of another expression *o* is said to be *undelimited* in *o* if:
> * *e* is *o*, or
> * *o* is an operator expression (infix, prefix or suffix operators) 
>     and *e* is undelimited in one of the operand expressions of *o*.

Then it is a compile-time error if a function expression is undelimited in the initializer expression.